tests/System.Buffers.Primitives.Tests/BufferReferenceTesting.cs

```diff
@@ -237,5 +243,37 @@ static void BufferLifetime(Buffer<byte> buffer)
             //{
             //}
         }
+
+        static void BufferBasics(ReadOnlyBuffer<byte> buffer)
+        {
+            var span = buffer.Span;
+            Assert.Equal(buffer.Length, span.Length);
+            Assert.True(buffer.IsEmpty || buffer.Length != 0);
```

Comment:
Why are we checking both buffer.IsEmpty and buffer.Length?
IsEmpty already does the check if Length == 0.
```C#
public bool IsEmpty => Length == 0;
```

https://github.com/dotnet/corefxlab/blob/master/src/System.Buffers.Primitives/System/Buffers/Buffer.cs#L61

> This test ensures that is buffer is not empty, then Length !=0

How does this test ensure that?

Assert.True(buffer.IsEmpty || X) => passes regardless of what X returns. If we have buffer.Length != 0, buffer.Empty should be false, No? Am I missing something here?